### PR TITLE
feat: add context_menu command 

### PR DESCRIPTION
**Pull request by eumarciel404**

### CODE REVIEW

**Review by eumarciel404 — COMMENTED** [review]
Lol, wtf what happened here?

File: packages/core/src/components/discord-command/DiscordCommand.ts [hunk 1]
```diff
@@ -86,15 +199,46 @@ export class DiscordCommand extends LitElement implements LightTheme {
 	@property({ type: Boolean, reflect: true, attribute: 'light-theme' })
 	public accessor lightTheme = false;
 
+	private readonly validButtonTypes = new Set(['user_command', 'message_command', 'slash_command']);
```
Comment (favna) [inline packages/core/src/components/discord-command/DiscordCommand.ts:202]:
```suggestion
	private readonly validCommandTypes = new Set(['user_command', 'message_command', 'slash_command']);
```

File: packages/core/src/components/discord-command/DiscordCommand.ts [hunk 2]
```diff
@@ -86,15 +199,46 @@ export class DiscordCommand extends LitElement implements LightTheme {
...
+
+	public checkType() {
+		if (this.type) {
+			if (typeof this.type !== 'string') {
+				throw new TypeError('DiscordCommand `type` prop must be a string.');
+			} else if (!this.validButtonTypes.has(this.type)) {
```
Comment (favna) [inline packages/core/src/components/discord-command/DiscordCommand.ts:208]:
```suggestion
			} else if (!this.validCommandTypes.has(this.type)) {
```

File: packages/core/src/components/discord-command/DiscordCommand.ts [hunk 3]
```diff
@@ -86,15 +199,46 @@ export class DiscordCommand extends LitElement implements LightTheme {
...
+				throw new RangeError("DiscordCommand `type` prop must be one of: 'context_menu', 'slash_command'");
```
Comment (favna) [inline packages/core/src/components/discord-command/DiscordCommand.ts:209]:
```suggestion
				throw new RangeError("DiscordCommand `type` prop must be one of: 'user_command', 'message_command', or 'slash_command'");
```

**Review by favna — CHANGES_REQUESTED** [review]

**Review by favna — APPROVED** [review]

**Pull request merged**